src/main/java/de/tum/in/www1/artemis/domain/LearningGoalProgress.java

```diff
@@ -0,0 +1,139 @@
+package de.tum.in.www1.artemis.domain;
+
+import java.io.Serial;
+import java.io.Serializable;
+import java.time.Instant;
+import java.util.Objects;
+
+import javax.persistence.*;
+
+import org.springframework.data.annotation.LastModifiedDate;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+/**
+ * This class models the 'progress' association between a user and a learning goal.
+ */
+@Entity
+@Table(name = "learning_goal_user")
+@EntityListeners(AbstractAuditingEntity.class)
+public class LearningGoalProgress {
```

Comment:
Why do you not extend DomainObject here?

Reply:
Because the object does not have a "normal" id, but a `@EmbeddedId`.